process/block/alteredaccounts/alteredAccountsProvider_test.go

```diff
@@ -0,0 +1,430 @@
+package alteredaccounts
+
+import (
+	"encoding/hex"
+	"math/big"
+	"strings"
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/data"
+	"github.com/ElrondNetwork/elrond-go-core/data/esdt"
+	"github.com/ElrondNetwork/elrond-go-core/data/indexer"
+	"github.com/ElrondNetwork/elrond-go-core/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/testscommon"
+	"github.com/ElrondNetwork/elrond-go/testscommon/state"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewAlteredAccountsProvider(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil shard coordinator", func(t *testing.T) {
+		t.Parallel()
+
+		args := getMockArgs()
+		args.ShardCoordinator = nil
+
+		aap, err := NewAlteredAccountsProvider(args)
+		require.Nil(t, aap)
+		require.Equal(t, ErrNilShardCoordinator, err)
+	})
+
+	t.Run("nil address converter", func(t *testing.T) {
+		t.Parallel()
+
+		args := getMockArgs()
+		args.AddressConverter = nil
+
+		aap, err := NewAlteredAccountsProvider(args)
+		require.Nil(t, aap)
+		require.Equal(t, ErrNilPubKeyConverter, err)
+	})
+
+	t.Run("nil accounts adapter", func(t *testing.T) {
+		t.Parallel()
+
+		args := getMockArgs()
+		args.AccountsDB = nil
+
+		aap, err := NewAlteredAccountsProvider(args)
+		require.Nil(t, aap)
+		require.Equal(t, ErrNilAccountsDB, err)
+	})
+
+	t.Run("nil marshalizer", func(t *testing.T) {
+		t.Parallel()
+
+		args := getMockArgs()
+		args.Marshalizer = nil
+
+		aap, err := NewAlteredAccountsProvider(args)
+		require.Nil(t, aap)
+		require.Equal(t, ErrNilMarshalizer, err)
+	})
+
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		args := getMockArgs()
+		aap, err := NewAlteredAccountsProvider(args)
+
+		require.NotNil(t, aap)
+		require.NoError(t, err)
+	})
+}
+
+func TestAlteredAccountsProvider_ExtractAlteredAccountsFromPool(t *testing.T) {
```

Comment:
Maybe add an extra test for an account which has multiple ESDT/NFT transfers with the same ID, but different nonces?

Reply:
added

Reply:
Which one is it, exactly?

Reply:
`testExtractAlteredAccountsFromPoolAddressHasMultipleNfts`